### PR TITLE
Introduce BookmarkManager

### PR DESCRIPTION
**Pull request by injectives**
Bookmark manager is a new opt-in feature that is enabled via driver `Config`.

It is responsible for providing an up-to-date set of bookmarks to driver on session creation and database data access.

Driver provides a default implementation via `BookmarkManagers.defaultManager`. The default implementation keeps track of a single local driver instance bookmarks only. However, it supports 2 extension points:
    - `updateListener` - a listener for new bookmarks
    - `bookmarkSupplier` - a supplier of additional bookmarks

It is possible to turn off bookmark manager on session level using `SessionConfig` even when it is enabled on the driver level.

Please note that this feature might have an impact on setups that do not manage the driver directly since there is a chance of getting a driver with bookmarks manager enabled.

Javadoc includes more general details.

### CODE REVIEW

File: testkit-backend/src/main/java/neo4j/org/testkit/backend/messages/requests/NewDriver.java [hunk 1]
```diff
@@ -286,6 +305,15 @@ public static class NewDriverBody {
...
         private List<String> trustedCertificates;
+        private BookmarkManagerConfig bookmarkManager;
+    }
+
+    @Setter
+    @Getter
+    public static class BookmarkManagerConfig {
+        private Map<String, Set<String>> initialBookmarks;
+        private boolean bookmarkSupplier;
+        private boolean notifyBookmarks;
```
Comment (bigmontz) [inline testkit-backend/src/main/java/neo4j/org/testkit/backend/messages/requests/NewDriver.java:316]: Testkit protocol will get updated in: https://github.com/neo4j-drivers/testkit/pull/506

This PR contains tests about the consumers and suppliers.

**Review by bigmontz — APPROVED** [review]
🔖 🕴🏼 

Please, you may have to do few tweaks in the backend for implementing the bookmark manager extension functions. 

https://github.com/neo4j-drivers/testkit/pull/506

File: testkit-backend/src/main/java/neo4j/org/testkit/backend/messages/requests/NewDriver.java (hunk 1 above)
Comment (injectives) [reply testkit-backend/src/main/java/neo4j/org/testkit/backend/messages/requests/NewDriver.java:316]: Thank you, it will be updated separately.

**Comment by injectives** [comment]
> 🔖 🕴🏼
> 
> Please, you may have to do few tweaks in the backend for implementing the bookmark manager extension functions.
> 
> [neo4j-drivers/testkit#506](https://github.com/neo4j-drivers/testkit/pull/506)

Thanks you. Sure, it will be updated separately.

**Comment by injectives** [comment]
Further surface update: https://github.com/neo4j/neo4j-java-driver/pull/1290

**Pull request merged**